src/app/api/collections/tree.py

```diff
@@ -1,44 +1,95 @@
-from __future__ import (  # Needed for recursive type annotation, can be dropped with Python>3.10
-    annotations,
-)
-
 from uuid import UUID
 
 from aiohttp import ClientSession
-from pydantic import BaseModel
+from elasticsearch_dsl.response import Response
+from glom import Coalesce, Iter, glom
+
+from app.api.collections.models import (
+    Collection,
+    CollectionAttribute,
+    CollectionTreeNode,
+)
+from app.api.collections.vocabs import tree_from_vocabs
+from app.core.config import ELASTIC_TOTAL_SIZE
+from app.elastic.dsl import qbool, qterm
+from app.elastic.elastic import add_base_match_filters
+from app.elastic.search import Search
+
+
+def build_portal_tree(
+    collections: list, root_noderef_id: UUID
+) -> list[CollectionTreeNode]:
+    lut = {str(root_noderef_id): []}
+
+    for collection in collections:
+        if collection.title:
+            portal_node = CollectionTreeNode(
+                noderef_id=collection.noderef_id,
+                title=collection.title,
+                children=[],
+            )
+
+            try:
+                lut[str(collection.parent_id)].append(portal_node)
+            except KeyError:
+                lut[str(collection.parent_id)] = [portal_node]
+
+            lut[str(collection.noderef_id)] = portal_node.children
+
+    return lut.get(str(root_noderef_id), [])
 
-from app.core.constants import COLLECTION_ROOT_ID
 
+def tree_query(node_id: UUID) -> Search:
```

Comment:
rename to `elastic_tree_query` ?

Reply:
Renamed to tree_search, in line with your other comment :)